lib/Service/FaceClusterAnalyzer.php

```diff
@@ -107,7 +107,8 @@ public function calculateClusters(string $userId) {
 				/** @var FaceDetection $detection */
 				$detection = $item[0];
 				$distance = new Euclidean();
-				if ($detection->getThreshold() > 0.0) {
+				// If threshold is larger than 0 and $clusterCentroid is not the null vector
+				if ($detection->getThreshold() > 0.0 && count(array_filter($clusterCentroid, fn($el) => $el !== 0.0)) > 0) {
```

Comment:
If the centroid could not be calculated, because there were zero detections in the cluster (see below), the centroid will be the null vector (`count(array_filter($clusterCentroid, fn($el) => $el !== 0.0)) > 0` checks if it is not the null vector). Then we don't want to apply the threshold either.

Reply:
Thanks for the explanations.